### ISSUE TITLE
Support paths with wasbs scheme for Azure

### ISSUE THREAD

**Issue by mrcnc** [lakekeeper/lakekeeper#417]
Currently only abfss paths are supported but it would be helpful to have storage profile support for wasbs paths as well.  This can support tables that have a default location defined with a wasbs scheme but can still be accessed via the Azure Data Lake Storage Gen2 REST APIs.

**Comment by c-thiel** [comment]
Good Point @mrcnc, we were a bit careful with this initially as most other Azure impls use abfss by default and Microsoft recommends to use abfss. Do you have an overview of how good the wasbs support is for different Iceberg clients?
I've seen a few discussions in pyiceberg https://github.com/apache/iceberg/issues/10127  - and potentially missing support for FileIO. Thats why we started with abfss

**Comment by mrcnc** [comment]
Currently the Java clients can use the FileIO via HDFS but I'll be working to add wasbs support to [ADLSFileIO](https://github.com/apache/iceberg/blob/d0a7ff915e6307e200a02ed76daff1a2fb363a3b/azure/src/main/java/org/apache/iceberg/azure/adlsv2/ADLSFileIO.java#L47) soon.   I'm not sure about the python clients, but  I believe lakekeeper would be using OpenDAL, so if we could translate a wasbs path into abfss in the catalog, it can still leverage the Azure Data Lake Storage Gen2 APIs through [azdls](https://nightlies.apache.org/opendal/opendal-docs-release-v0.46.0/docs/services/azdls/).

**Comment by c-thiel** [comment]
Now that https://github.com/apache/iceberg/pull/11830/files is merged, we are going to implement it for Lakekeeper too :)
